cartoframes/client/data_obs_client.py

```diff
@@ -0,0 +1,692 @@
+                bounds = 'ST_MakeEnvelope(-180.0, -85.0, 180.0, 85.0, 4326)'
+        elif isinstance(region, collections.Iterable):
+            if len(region) != 4:
+                raise ValueError(
+                    '`region` should be a list of the geographic bounds of a '
+                    'region in the following order: western longitude, '
+                    'southern latitude, eastern longitude, and northern '
+                    'latitude. For example, Switerland fits in '
+                    '``[5.9559111595,45.8179931641,10.4920501709,'
+                    '47.808380127]``.')
+            bounds = ('ST_MakeEnvelope({0}, {1}, {2}, {3}, 4326)').format(
+                *region)
+        elif region is None:
+            bounds = 'ST_MakeEnvelope(-180.0, -85.0, 180.0, 85.0, 4326)'
+        else:
+            raise ValueError('`region` must be a str, a list of two lng/lat '
+                             'pairs, or ``None`` (which defaults to the '
+                             'world)')
+        if include_nonclipped:
+            clipped = None
+        else:
+            clipped = (r"(geom_id ~ '^us\.census\..*_clipped$' OR "
+                       r"geom_id !~ '^us\.census\..*')")
+
+        if boundary is None:
+            regionsearch = locals().get('regionsearch')
+            filters = ' AND '.join(r for r in [regionsearch, clipped] if r)
+            query = utils.minify_sql((
+                'SELECT *',
+                'FROM OBS_GetAvailableGeometries(',
+                '  {bounds}, null, null, null, {timespan})',
+                '{filters}')).format(
+                    bounds=bounds,
+                    timespan=utils.pgquote(timespan),
+                    filters='WHERE {}'.format(filters) if filters else '')
+            return self._fetch(query, decode_geom=True)
+
+        query = utils.minify_sql((
+            'SELECT the_geom, geom_refs',
+            'FROM OBS_GetBoundariesByGeometry(',
+            '    {bounds},',
+            '    {boundary},',
+            '    {time})', )).format(
+                bounds=bounds,
+                boundary=utils.pgquote(boundary),
+                time=utils.pgquote(timespan))
+        return self._fetch(query, decode_geom=decode_geom)
+
+    def discovery(self, region, keywords=None, regex=None, time=None,
+                  boundaries=None, include_quantiles=False):
+        """Discover Data Observatory measures. This method returns the full
+        Data Observatory metadata model for each measure or measures that
+        match the conditions from the inputs. The full metadata in each row
+        uniquely defines a measure based on the timespan, geographic
+        resolution, and normalization (if any). Read more about the metadata
+        response in `Data Observatory
+        <https://carto.com/docs/carto-engine/data/measures-functions/#obs_getmetaextent-geometry-metadata-json-max_timespan_rank-max_score_rank-target_geoms>`__
```

Comment:
This URL should be updated to: https://carto.com/developers/data-observatory/reference/#discovery-functions